machinery/topology.go

```diff
@@ -88,7 +89,7 @@ func NewTopology(options ...TopologyOptionsFunc) *Topology {
 		return t
 	})
 
-	graph := dot.NewGraph(dot.Directed)
+	graph := dot.NewGraph(dot.Directed, dot.Strict)
```

Comment:
Added the `dot.Strict` to ensure there are more that one connection between two nodes.

Reply:
I would have to check the algorithm to ensure it is not need. 

First reason I set it was the graph was just harder to follow when debugging one of the unit tests. But I think a valid reason to why we should not be allowing it is when you come to attached policy. Without it would mean a policy can be attached to the same route multiply times and depending on the like functions this night not be happening at the same time. While I don't know if this could cause a problem with defaults and overrides I think it would be best not to allow the double linking until there is a use case for it.